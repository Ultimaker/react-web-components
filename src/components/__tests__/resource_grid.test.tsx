// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow } from 'enzyme';

// component
import ResourceGrid, { ResourceGridProps } from '../../components/resource_grid';

describe('The ResourceGrid component', () => {
    let props: ResourceGridProps;
    let wrapper;

    beforeEach(() => {
        props = {
            emptyStateText: 'Get started by connecting your devices to the cloud',
            emptyStateImageUrl: '../../images/logobot-placeholder-dark.svg',
            emptyTileImageUrl: '../../images/logobot-placeholder-dark.svg',
            emptyStateImageAlt: 'My devices',
            emptyTileImageAlt: 'My devices',
            createButtonText: 'Add printers',
            onCreate: jest.fn(),
        };
        wrapper = shallow(<ResourceGrid {...props} />);
    });

    it('should render the empty state', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a grid', () => {
        wrapper.setProps({ children: <div>Test item</div> });
        expect(wrapper).toMatchSnapshot();
    });
});
