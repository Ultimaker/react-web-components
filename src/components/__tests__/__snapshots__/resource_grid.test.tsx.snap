// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ResourceGrid component should render a grid 1`] = `
<div>
  <Grid
    align="center"
    gutter="md"
    position="top"
  >
    <GridItem
      breakpoint="xs"
      key=".0"
      layoutWidth="fit"
    >
      <div>
        Test item
      </div>
    </GridItem>
    <GridItem
      breakpoint="xs"
      layoutWidth="fit"
    >
      <ResourceEmptyTile
        createButtonText="Add printers"
        imageAlt="My devices"
        imageUrl="../../images/logobot-placeholder-dark.svg"
        onCreateHandler={[MockFunction]}
      />
    </GridItem>
  </Grid>
</div>
`;

exports[`The ResourceGrid component should render the empty state 1`] = `
<ResourceEmptyState
  createButtonText="Add printers"
  imageAlt="My devices"
  imageUrl="../../images/logobot-placeholder-dark.svg"
  onCreateHandler={[MockFunction]}
  text="Get started by connecting your devices to the cloud"
/>
`;
